From fb78b2617e70e58e03e0d50e674758fdd2a80d45 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 2 Nov 2023 10:39:11 +0000
Subject: [PATCH] dts: bcm2712-rpi-5-b: Create some dummy nodes

The kernel now treats multiple fragments targeting the same node as an
error. For this reason, it is important that labels created just for
compatibility with other systems (e.g. i2c0if and i2c0mux) are
attached to unique nodes, not just tacked onto existing nodes.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2712-rpi-5-b.dts | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

--- a/arch/arm/boot/dts/bcm2712-rpi-5-b.dts
+++ b/arch/arm/boot/dts/bcm2712-rpi-5-b.dts
@@ -157,6 +157,12 @@
 	dummy: dummy {
 		// A target for unwanted overlay fragments
 	};
+
+
+	// A few extra labels to keep overlays happy
+
+	i2c0if: i2c0if {};
+	i2c0mux: i2c0mux {};
 };
 
 rp1_target: &pcie2 {
@@ -243,11 +249,6 @@ aux: &dummy {};
 
 #include "bcm2712-rpi.dtsi"
 
-// A few extra labels to keep overlays happy
-
-i2c0if: &rp1_gpio {};
-i2c0mux: &rp1_gpio {};
-
 i2c_csi_dsi0: &i2c6 { // Note: This is for MIPI0 connector only
 	pinctrl-0 = <&rp1_i2c6_38_39>;
 	pinctrl-names = "default";
